docs-user/bunny.md

```diff
@@ -0,0 +1,272 @@
+# Case Study
+## 2D canvas and worker messaging
+
+The following article is a case study in using the profiler to identify a performance issue, and then fixing it for a 4 times performance improvement. This improvement moves the frame rate from a fairly slow 15fps to a smooth 60fps.
+
+## The project description
+
+![A picture of a 3d bunny rabbit model rendered using small squares.](./images/bunny-analysis/bunny.png)
+
+The project is a website that takes user's JavaScript code, and runs it to produce a visualization. The user's code only has access to the function `rect(color, x, y, width, height)`, which draws a rectangle to the screen. For the implementation, the website posts the user's code to a sandboxed iframe. The iframe has a `<canvas>` element that is rendered to via the [`CanvasRenderingContext2D`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D) API. The user's code is evaluated in a WebWorker, and then the results of `rect()` are posted back to the iframe's code.
+
+In a simplified example, the `worker.js` would run something like:
+
+```js
+// Evaluate the user's code string:
+eval(`
+rect('#fff', 10, 10, 3, 3);
+rect('#fff', 20, 20, 3, 3);
+rect('#fff', 30, 30, 3, 3);
+rect('#fff', 40, 40, 3, 3);
+rect('#fff', 50, 50, 3, 3);
+`);
+
+// worker.js would post something like this to the iframe:
+self.postMessage({
+  color: ['#fff', '#fff', '#fff', '#fff', '#fff'],
+  x: [10, 20, 30, 40, 50],
+  y: [10, 20, 30, 40, 50],
+  w: [3, 3, 3, 3, 3],
+  h: [3, 3, 3, 3, 3],
+});
+```
```

Comment:
The colors from the syntax highlighting are not so nice for me; especially the comments are very pale compared to the background. The text is in general very pale, I miss some contrast.

![image](https://user-images.githubusercontent.com/454175/40552055-f5456cf2-603e-11e8-8bc6-aba414f4babb.png)

Reply:
<img width="791" alt="image" src="https://user-images.githubusercontent.com/1588648/40686830-3f307c22-635e-11e8-89fa-d5a77d4e214f.png">

That's odd, it's much clearer on my screen.

I filed: https://github.com/devtools-html/perf.html/issues/1052

Reply:
Perhaps this is OS difference?